### PR TITLE
Run3-hcx308 Rearrange the files to remove .h files from SimCalorimetry/CastorSim/src

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

Rearrange the files to remove .h files from SimCalorimetry/CastorSim/src

#### PR validation:

Use the runTheMatrix test workflows 

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by civanch** [comment]
+1

File: SimCalorimetry/CastorSim/plugins/CastorDigiAnalyzer.cc [hunk 1]
```diff
@@ -1,7 +1,87 @@
...
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
 #include "FWCore/MessageLogger/interface/MessageLogger.h"
-#include "SimCalorimetry/CastorSim/plugins/CastorDigiAnalyzer.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "SimCalorimetry/CaloSimAlgos/interface/CaloHitAnalyzer.h"
+#include "SimCalorimetry/CaloSimAlgos/interface/CaloValidationStatistics.h"
+#include "SimCalorimetry/CaloSimAlgos/interface/CaloHitAnalyzer.h"
```
Comment (perrotta) [inline SimCalorimetry/CastorSim/plugins/CastorDigiAnalyzer.cc:15]: Duplicate of line 13

File: SimCalorimetry/CastorSim/plugins/CastorDigiAnalyzer.h [hunk 2]
```diff
@@ -9,8 +9,8 @@
 #include "FWCore/Utilities/interface/InputTag.h"
```
Comment (perrotta) [inline SimCalorimetry/CastorSim/plugins/CastorDigiAnalyzer.h:9]: This file was included in CastorDigiAnalyzer.cc, and it should therefore get removed

File: SimCalorimetry/CastorSim/plugins/CastorDigiProducer.h [hunk 3]
```diff
@@ -11,14 +11,14 @@
 #include "Geometry/Records/interface/CaloGeometryRecord.h"
```
Comment (perrotta) [inline SimCalorimetry/CastorSim/plugins/CastorDigiProducer.h:11]: This file was included in CastorDigiProducer.cc, and it should therefore get removed

File: SimCalorimetry/CastorSim/plugins/CastorHitAnalyzer.h [hunk 4]
```diff
@@ -7,8 +7,8 @@
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
```
Comment (perrotta) [inline SimCalorimetry/CastorSim/plugins/CastorHitAnalyzer.h:7]: This file was included in CastorHitAnalyzer.cc, and it should therefore get removed

**Comment by civanch** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**